scripts/bootstrap

```diff
@@ -0,0 +1,200 @@
+    fi
+}
+
+main() {
+    # Allow changing the default branch
+    [[ -n "$TOLTEC_BRANCH" ]] || TOLTEC_BRANCH="stable"
+
+    # Select source for remarkable_entware repo
+    if [ -z "$REMARKABLE_ENTWARE_REPO_AUTHOR" ]; then
+        # Use LinusCDE's fork for the time beeing
+        # TODO: Switch to a more official repo when it exists
+        REMARKABLE_ENTWARE_REPO_AUTHOR=LinusCDE
+    fi
+
+    # Check status of installation and ensure installed
+    if [ -d /home/root/.entware ]; then
+        if [[ -d /opt ]] && files="$(ls -qA -- /opt)" && [[ -n $files ]]; then
+            # Entware is installed and active
+            log INFO "Entware is installed and active"
+        else
+            # Installed but not active
+            log INFO "Entware is installed but not reenabled"
+            entware_reenable
+        fi
+    else # = not found: /home/root/.entware
+        # Not installed at all
+        log INFO "Entware is not installed"
+        entware_install
+    fi
+
+    # Ensure opkg is in PATH
+    ensure_opkg_available
+    # Ensure wget is installed as opkg package
+    ensure_opt_wget_and_ca_certs_installed
+    # Add toltec repo if mising
+    ensure_toltec_repo_added
+
+    if [ $# -ge 1 ]; then
+        install_packages "$@" || throw "Failed to install requested packages"
+    fi
+}
+
+entware_install() {
+    bootstrap_wget || throw "bootstrap_wget() failed!"
+    (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_install.sh" | sh) \
+        || throw "Failed to download and/or install remarkable_entware."
+    log INFO "Entware was installed successfully."
+}
+
+entware_reenable() {
+    bootstrap_wget || throw "bootstrap_wget() failed!"
+    (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_reenable.sh" | sh) \
+        || throw "Failed to download and/or install remarkable_entware."
+    log INFO "Entware was reenabled successfully."
+}
+
+ensure_opkg_available_from_bashrc() {
+    # Run by ensure_opkg_available
+    if ! grep '.*PATH=.*/opt/bin.*' /home/root/.bashrc | grep -v '^#' >/dev/null; then
+        # Path is not in .bashrc
+        echo -e '\n# Path added by bootstrap.sh' >> /home/root/.bashrc
+        echo 'PATH="/opt/bin:/opt/sbin:$PATH"' >> /home/root/.bashrc
+        log WARN '/opt/bin and /opt/sbin were not added to $PATH in ~/.bashrc. Fixed automatically.'
+    fi
+}
+
+ensure_opkg_available() {
+    if command -v opkg &> /dev/null; then # opkg command not found
+        export PATH="/opt/bin/:/opt/sbin/:$PATH"
+        messages="$(ensure_opkg_available_from_bashrc)"
+        if [[ -n $messages ]]; then
+            echo "$messages" # Print the caught output
+            log WARN "Please re-connect to your reMarkable after this" \
+                     "script is finished. Otherwise opkg and other" \
+                     "installed binaries won't be recognized!"
```

Comment:
Isn’t this script supposed to run on the reMarkable? How can this message be shown if the connection to the reMarkable has been lost?

Reply:
It's instructing the user to reconnect after it's done running. Not that the user will lose connection.

Reply:
I'm not entirely sure where see the problem with where it runs. The bootstrap script is always meant to run only on the reMarkable and don't make any assumptions on the users computer os that way.

That said, checking whether the function echo'ed something is still dirty. I also should clean up or reduce or entirely remove `ensure_opkg_available()` and `ensure_opkg_available_from_bashrc()` since they're only run once anyway. Probably a case of YAGNI.

Reply:
> It's instructing the user to reconnect after it's done running. Not that the user will lose connection.

Yes. (I didn't saw the comment when I wrote the previous one.)

Is there a way to apply the updated $PATH to the users bash directly. But aside from spawning a child bash with exec I didn't find a good way to do that. So this message was my only solution.

Reply:
`export PATH=$PATH:...`?

Reply:
Only if you source the script. While we could do that in the one liner if someone downloads the script and runs it manually (which we shouldn't discourage imo), it wouldn't work since the export is applied the the child process the script is run in.

Reply:
When running the script source it instead of calling it and you can export variables.

Reply:
We could use that for the bootstrap command. But some people may wonder why. Also should people download and execute the script themselves, it wouldn't work.  
Is there a way to detect whether a script was sourced?

Reply:
https://stackoverflow.com/a/2687092